# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/shamiquekhan/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

In [1]:
# Setup: navigate to repo root
import os, sys
IN_COLAB = "google.colab" in sys.modules
if IN_COLAB:
    import subprocess
    if not os.path.isdir("flyrank-ml-internship"):
        subprocess.run(["git", "clone", "--depth", "1", "https://github.com/shamiquekhan/flyrank-ml-internship", "flyrank-ml-internship"], check=True)
    os.chdir("flyrank-ml-internship")
else:
    while not os.path.isdir("data/raw") and os.getcwd() != "/":
        os.chdir("..")
print("Working dir:", os.getcwd())

Working dir: /home/shamique/flyrank/ml1/flyrank-ml-internship


## 1. My lane (or freestyle) and why

I selected the **Refresh / Content Opportunity Scoring** lane because it addresses a practical business problem: content teams have limited capacity and need a data-driven way to decide which pages to review first. The starter dataset already contains relevant signals — impressions, trend direction, CTR, content age, and engagement metrics — making it feasible to build a transparent ranking system. This lane interests me because it connects directly to how ML supports human decision-making: not replacing reviewers, but helping them spend time where it matters most.

## 2. The question: decision, action, cost of a wrong call

**Search question:** Can I predict which pages are most likely to need a content refresh, using observable page-level signals, so reviewers can prioritize their limited capacity?

**Unit of analysis:** One webpage (one row per content_id in the starter dataset).

**Model output:** A ranked priority score (0-100) for each page, indicating how strongly it qualifies for review.

**Decision:** Content operations specialists deciding which pages to review for refresh, expansion, or optimization.

**Action:** They review the highest-ranked pages first — rewrite, update, merge, or monitor based on the reason codes I provide with each score.

**Cost of a wrong prediction:**
- False positives (flagging a page that does not need refresh) waste reviewer time and erode trust in my system.
- False negatives (missing a page that genuinely needs attention) can lead to continued traffic decline on valuable content, missing opportunities to recover visibility.

## 3. Quick look at the data (2-3 real numbers)

*I load the starter CSV below to compute real statistics from the actual dataset.*

In [2]:
# Load the starter dataset and show real numbers
import pandas as pd, numpy as np
df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print(f"Rows (pages): {len(df):,}")
print(f"Features: {df.shape[1]}")
print()

declining_pct = (df["trend_direction"].str.lower() == "down").mean() * 100
print(f"Declining pages: {declining_pct:.1f}% of all pages")
print()

visible = (df["impressions_90d"] >= 100).sum()
print(f"Pages with >= 100 impressions (visible demand): {visible:,} ({visible/len(df)*100:.1f}%)")
print()

stale_visible = ((df["days_since_last_update"] >= 180) & (df["impressions_90d"] >= 500)).sum()
print(f"Pages that are both stale (>=180 days) AND visible (>=500 impressions): {stale_visible:,}")
print("This intersection represents clear refresh candidates — the lane has real inventory to rank.")

Rows (pages): 30,000
Features: 44

Declining pages: 54.2% of all pages

Pages with >= 100 impressions (visible demand): 22,006 (73.4%)

Pages that are both stale (>=180 days) AND visible (>=500 impressions): 17
This intersection represents clear refresh candidates — the lane has real inventory to rank.


### Why these statistics support my lane choice

I observe that the dataset contains 30,000 pages with 44 features — enough observations and signal variety for me to build a meaningful ranking model. Over half the pages (54.2%) are currently declining, giving me a substantial positive class to learn from. Critically, I see pages that are both stale (not updated in 6+ months) and still receiving significant traffic — these are exactly the kind of pages my refresh scoring system should prioritize. This confirms the lane has real inventory for me to work with.

## 4. Careful words: what I can and can't claim

My exploratory analysis does not establish causation. I observe relationships between page-level signals and performance, but I can only report measured associations — I cannot claim that refreshing a page causes recovery without an experimental design.

The dataset may contain biases: it covers only clients with existing FlyRank tracking, and the anonymization process may obscure some nuances. My starter slice (30,000 rows) is a sample of a larger warehouse; my results here may differ on the full release.

My model can assist human reviewers by ranking candidates, but it should not replace human judgment. I frame every claim as observed, measured, directional, and decision-support — never as proof of Google's algorithm or as guaranteed outcomes of specific actions.

## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.